nvflare/fuel/f3/streaming/byte_streamer.py

```diff
@@ -71,17 +82,162 @@ def __init__(
         self.offset_ack = 0
         self.secure = secure
         self.optional = optional
+        self.stopped = False
+
+        self.stream_future = StreamFuture(self.sid)
+        self.stream_future.set_size(stream.get_size())
+
+        self.window_size = CommConfigurator().get_streaming_window_size(STREAM_WINDOW_SIZE)
+        self.ack_wait = CommConfigurator().get_streaming_ack_wait(STREAM_ACK_WAIT)
 
     def __str__(self):
         return f"Tx[SID:{self.sid} to {self.target} for {self.channel}/{self.topic}]"
 
+    def send_loop(self):
```

Comment:
For this underlying change, is it low-risk? If QA can't test 2.5.1 thoroughly, should we move this PR to main?

Reply:
The risk is pretty low but it's a good idea to merge the change into main. All the bugs we discovered are already fixed in 2.5.

Reply:
Created https://github.com/NVIDIA/NVFlare/pull/2960 for main branch.